### PR TITLE
Add IconLibrary Source

### PR DESCRIPTION
**Pull request by gettinToasty**

### CODE REVIEW

File: app/components-react/windows/IconLibraryProperties.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+  function lifecycle() {
+    if (source) {
+      const settings = source.getPropertiesManagerSettings();
+      if (settings.folder) {
+        const { folder, activeIcon } = settings;
+        setFolderPath(folder);
+
+        fs.readdir(folder, (err: Error, files: string[]) => {
+          setFolderImages(files.map((file: string) => path.join(folder, file)));
```
Comment (andycreeth) [inline app/components-react/windows/IconLibraryProperties.tsx:32]: Should handle the error case here.  When accessing the filesystem, errors are common and to be expected.

File: app/components-react/windows/IconLibraryProperties.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        setSelectedIcon(activeIcon);
+      }
+    }
+  }
+
+  function selectFolder(folder: string) {
+    if (!source) return;
+    setFolderPath(folder);
+
+    fs.readdir(folder, (err: Error, files: string[]) => {
```
Comment (andycreeth) [inline app/components-react/windows/IconLibraryProperties.tsx:44]: Same comment regarding handling errors.

**Review by andycreeth — CHANGES_REQUESTED** [review]
Looks good!  Just needs the error handling.

**Review by holiber — APPROVED** [review]
Nice

**Review by andycreeth — APPROVED** [review]

**Pull request merged**